proxy/http/HttpTunnel.cc

```diff
@@ -956,6 +956,9 @@ HttpTunnel::producer_run(HttpTunnelProducer *p)
       // Start the writes now that we know we will consume all the initial data
       c->write_vio = c->vc->do_io_write(this, c_write, c->buffer_reader);
       ink_assert(c_write > 0);
+      if (c->write_vio == nullptr) {
```

Comment:
Actually, I'm a bit surprised that we haven't had this check and haven't seen any problems. It looks like this case could happen with HTTP/1.1. Probably, Http2Stream easily falls down this case because of the life cycle.